internal/internal_workflow_client.go

```diff
@@ -1866,42 +1881,18 @@ func (w *workflowClientInterceptor) UpdateWorkflow(
 	ctx context.Context,
 	in *ClientUpdateWorkflowInput,
 ) (WorkflowUpdateHandle, error) {
-	argPayloads, err := w.client.dataConverter.ToPayloads(in.Args...)
+	req, err := w.convertToUpdateWorkflowRequest(ctx, in)
 	if err != nil {
 		return nil, err
 	}
-	header, err := headerPropagated(ctx, w.client.contextPropagators)
-	if err != nil {
-		return nil, err
-	}
-	desiredLifecycleStage := updateLifeCycleStageToProto(in.WaitForStage)
+
 	var resp *workflowservice.UpdateWorkflowExecutionResponse
 	for {
 		var err error
 		resp, err = func() (*workflowservice.UpdateWorkflowExecutionResponse, error) {
 			grpcCtx, cancel := newGRPCContext(ctx, grpcTimeout(pollUpdateTimeout), grpcLongPoll(true), defaultGrpcRetryParameters(ctx))
 			defer cancel()
-			wfexec := &commonpb.WorkflowExecution{
-				WorkflowId: in.WorkflowID,
-				RunId:      in.RunID,
-			}
-			return w.client.workflowService.UpdateWorkflowExecution(grpcCtx, &workflowservice.UpdateWorkflowExecutionRequest{
```

Comment:
Note that `workflowservice.UpdateWorkflowExecutionRequest` is only created once, outside of the for loop, now.